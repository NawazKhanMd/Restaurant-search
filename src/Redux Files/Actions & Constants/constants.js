export const home_action_types = {
    loading:'loading',
    get_movies_success:'get_movies_success',
    no_movies_found:'no_movies_found',
    update_fav_movies:'update_fav_movies',
    get_movie_details:'get_movie_details'
} 