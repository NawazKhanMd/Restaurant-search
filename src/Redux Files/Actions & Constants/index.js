export * from './home_actions';