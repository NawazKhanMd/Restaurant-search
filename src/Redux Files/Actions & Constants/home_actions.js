import { home_action_types as action_types } from '../Actions & Constants/constants'
import {restApiUrls,RequestAPI} from '../../Utils/fetchInterceptor'
export const getCities = () => (dispatch) => {
  dispatch(toogleLoader(true))
  let fetchObj = {
    url: restApiUrls.CitiesURL,
    method: 'GET'
}
return RequestAPI(fetchObj).then(resp => {
  dispatch(toogleLoader(false))
    if (resp.type == 'success') {
        dispatch({
          type:action_types.get_cities_success,
          payload:resp.data
        })
    } else {
      dispatch(toogleLoader(false))
    }
})
}
export const toogleLoader = (flag) => (dispatch)=>{
  dispatch({
    type:action_types.loading,
    flag:flag
  })
}
export const filterRestaurants= (address)=>(dispatch)=>{
  dispatch({
    type:action_types.filter_restaurants,
    address:address
  })
}

export const getRestaurants = (City) => (dispatch) => {
  dispatch(toogleLoader(true))
  let fetchObj = {
    url: restApiUrls.RestaurantURL.replace('#{CITY}#',City),
    method: 'GET'
}
return RequestAPI(fetchObj).then(resp => {
  dispatch(toogleLoader(false))
    if (resp.type == 'success') {
        dispatch({
          type:action_types.get_restaurants_success,
          payload:resp.data
        })
    } else {
      dispatch(toogleLoader(false))
    }
})
}

