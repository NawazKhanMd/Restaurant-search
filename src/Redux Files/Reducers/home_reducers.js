import { home_action_types as atn_typ } from '../Actions & Constants/constants'
export const initialState = {
    loading: false,
    movies: [],
    favMovies: localStorage.getItem('favMovies') == null ? [] : JSON.parse(localStorage.getItem('favMovies')),
    selectedMovie: {}
};


export const home = (state = initialState, action) => {

    switch (action.type) {
        case atn_typ.get_movies_success:
            state.noMoviesFound = false;
            if (action.pageNumber == 1) {
                state.movies = [...action.payload.Search]
            } else {
                state.movies = [...state.movies, ...action.payload.Search]
            }
            return { ...state }
        case atn_typ.no_movies_found:
            state.noMoviesFound = true;
            state.movies = [];
            return { ...state }
        case atn_typ.update_fav_movies:
            let indexOfFavMov = state.favMovies.findIndex(num => num == action.payload.imdbID)
            if (indexOfFavMov == -1) {
                state.favMovies.push(action.payload.imdbID)
            } else {
                state.favMovies.splice(indexOfFavMov, 1)
            }
            localStorage.setItem('favMovies', JSON.stringify(state.favMovies));
            state.movies = [...state.movies]
            return { ...state }
        case atn_typ.get_movie_details:
            state.selectedMovie = { ...action.payload }
            return { ...state }
        case atn_typ.loading:
            state.loading = action.flag
            return { ...state }
        default:
            return state
    }
}
