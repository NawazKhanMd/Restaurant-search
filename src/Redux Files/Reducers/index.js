import { combineReducers } from "redux";
import {home} from './home_reducers'
export default combineReducers({
    home
});