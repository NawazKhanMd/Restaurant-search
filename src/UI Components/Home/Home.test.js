import React from 'react';
import { Provider } from 'react-redux'
import { mount, shallow, configure } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import { Home } from './index'
import { initialState } from '../../Redux Files/Reducers/home_reducers'
import Adapter from 'enzyme-adapter-react-16';
import { home_action_types as atn_typs } from '../../Redux Files/Actions & Constants/constants';
import toJson from 'enzyme-to-json';
import TestRenderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const mockStore = configureMockStore([thunk]);

describe('Home tests', () => {
  test('Home renders correctly enzyme', () => {
    const store = mockStore({
      home: { ...initialState }
    });
    const wrapper = mount(
      <Provider store={store}>
        <Home />
      </Provider>
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('Home should render 1 Movie', () => {
    const store = mockStore({
      home: { ...initialState }
    });
    const wrapper = mount(
      <Provider store={store}>
        <Home />
      </Provider>
    )
    mount(<Provider store={store}> <Home /> </Provider>)
    const data = {
      "Search": [
        {
          "Title": "Spider-Man",
          "Year": "2002",
          "imdbID": "tt0145487",
          "Type": "movie",
          "Poster": "https://m.media-amazon.com/images/M/MV5BZDEyN2NhMjgtMjdhNi00MmNlLWE5YTgtZGE4MzNjMTRlMGEwXkEyXkFqcGdeQXVyNDUyOTg3Njg@._V1_SX300.jpg"
        }, { "Title": "The Amazing Spider-Man", "Year": "2012", "imdbID": "tt0948470", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BMjMyOTM4MDMxNV5BMl5BanBnXkFtZTcwNjIyNzExOA@@._V1_SX300.jpg" }, { "Title": "Spider-Man 2", "Year": "2004", "imdbID": "tt0316654", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BMzY2ODk4NmUtOTVmNi00ZTdkLTlmOWYtMmE2OWVhNTU2OTVkXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg" }, { "Title": "Spider-Man: Homecoming", "Year": "2017", "imdbID": "tt2250912", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BNTk4ODQ1MzgzNl5BMl5BanBnXkFtZTgwMTMyMzM4MTI@._V1_SX300.jpg" }, { "Title": "Spider-Man 3", "Year": "2007", "imdbID": "tt0413300", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BYTk3MDljOWQtNGI2My00OTEzLTlhYjQtOTQ4ODM2MzUwY2IwXkEyXkFqcGdeQXVyNTIzOTk5ODM@._V1_SX300.jpg" }, { "Title": "The Amazing Spider-Man 2", "Year": "2014", "imdbID": "tt1872181", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BOTA5NDYxNTg0OV5BMl5BanBnXkFtZTgwODE5NzU1MTE@._V1_SX300.jpg" }, { "Title": "Spider-Man: Into the Spider-Verse", "Year": "2018", "imdbID": "tt4633694", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BMjMwNDkxMTgzOF5BMl5BanBnXkFtZTgwNTkwNTQ3NjM@._V1_SX300.jpg" }, { "Title": "Spider-Man: Far from Home", "Year": "2019", "imdbID": "tt6320628", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BMGZlNTY1ZWUtYTMzNC00ZjUyLWE0MjQtMTMxN2E3ODYxMWVmXkEyXkFqcGdeQXVyMDM2NDM2MQ@@._V1_SX300.jpg" }, { "Title": "Along Came a Spider", "Year": "2001", "imdbID": "tt0164334", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BOTVlY2VhMWEtYmRlOC00YWVhLWEzMDktZWJlYzNiMWJmZTIwXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg" }, { "Title": "Spider", "Year": "2002", "imdbID": "tt0278731", "Type": "movie", "Poster": "https://m.media-amazon.com/images/M/MV5BMmY4OGRmNWMtNmIyNS00YWQ5LWJmMGUtMDI3MWRlMmQ0ZDQzL2ltYWdlXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg" }
      ], "totalResults": "490", "Response": "True"
    }
    store.dispatch({
      type: atn_typs.get_movies_success,
      payload: { ...data }
    });
    const actions = store.getActions();
    expect(actions[actions.length - 1].type).toEqual(atn_typs.get_movies_success);
    expect(actions[actions.length - 1].payload).toEqual(data);
  })
})
