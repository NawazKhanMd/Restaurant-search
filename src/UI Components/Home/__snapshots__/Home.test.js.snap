// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests Home renders correctly enzyme 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Home>
    <div
      className="home-container"
    >
      <div
        className="fav-movies-btn"
      >
        <FontAwesomeIcon
          border={false}
          className="small-fav-button"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f004",
                "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
              ],
              "iconName": "heart",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart fa-w-16 small-fav-button"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </div>
      <div
        className="input-container  dontMove"
      >
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <ForwardRef
            className="justify-content-md-center margin0"
            noGutters={false}
          >
            <div
              className="justify-content-md-center margin0 row"
            >
              <Col
                lg={12}
                md={12}
                xs={10}
              >
                <div
                  className="col-lg-12 col-md-12 col-10"
                >
                  <div
                    className="input-card"
                  >
                    <input
                      name="Search"
                      placeholder="Search for a movie"
                      type="text-input"
                    />
                    <button
                      className="search-btn"
                      type="submit"
                    >
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              512,
                              512,
                              Array [],
                              "f002",
                              "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                            ],
                            "iconName": "search",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        swapOpacity={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-search fa-w-16 "
                          data-icon="search"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                    </button>
                  </div>
                </div>
              </Col>
            </div>
          </ForwardRef>
        </form>
      </div>
      <div
        className="slide-panel  slideDown"
      >
        <div
          className="slide-panel-imageBoard"
          style={
            Object {
              "backgroundImage": "url(undefined)",
            }
          }
        >
          <FontAwesomeIcon
            border={false}
            className="close-button"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  352,
                  512,
                  Array [],
                  "f00d",
                  "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                ],
                "iconName": "times",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-times fa-w-11 close-button"
              data-icon="times"
              data-prefix="fas"
              focusable="false"
              onClick={[Function]}
              role="img"
              style={Object {}}
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
          <FontAwesomeIcon
            border={false}
            className="small-un-fav-button"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f004",
                  "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
                ],
                "iconName": "heart",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-heart fa-w-16 small-un-fav-button"
              data-icon="heart"
              data-prefix="fas"
              focusable="false"
              onClick={[Function]}
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
          <ForwardRef
            className="margin0"
            noGutters={false}
          >
            <div
              className="margin0 row"
            >
              <Col
                lg={2}
                md={2}
                xs={2}
              >
                <div
                  className="col-lg-2 col-md-2 col-2"
                >
                  <Card
                    body={false}
                  >
                    <div
                      className="card"
                    >
                      <CardImg
                        variant="top"
                      >
                        <img
                          className="card-img-top"
                        />
                      </CardImg>
                    </div>
                  </Card>
                </div>
              </Col>
            </div>
          </ForwardRef>
        </div>
        <div
          className="slide-panel-detailsBoard"
        >
          <ForwardRef
            className="margin0"
            noGutters={false}
          >
            <div
              className="margin0 row"
            >
              <Col
                lg={
                  Object {
                    "offset": 3,
                    "span": 9,
                  }
                }
                md={
                  Object {
                    "offset": 3,
                    "span": 9,
                  }
                }
                xs={12}
              >
                <div
                  className="col-lg-9 col-md-9 col-12 offset-lg-3 offset-md-3"
                >
                  <div
                    className="one-line"
                  >
                    <h3>
                       
                    </h3>
                     
                    <p>
                      (
                      )
                    </p>
                     
                  </div>
                  <p>
                    Director :  
                    <b />
                  </p>
                  <p>
                    Writer :  
                    <b />
                  </p>
                  <p>
                    IMDB Rating: 
                    <b />
                    , Released: 
                    <b />
                  </p>
                  <div
                    className="one-line"
                  >
                    <p>
                      Cast: 
                    </p>
                    <p>
                      <b />
                    </p>
                  </div>
                  <div
                    className="one-line"
                  >
                    <p>
                      Plot: 
                    </p>
                    <p>
                      <b />
                    </p>
                  </div>
                </div>
              </Col>
            </div>
          </ForwardRef>
        </div>
      </div>
      <div
        className="movies-container"
        onScroll={[Function]}
      >
        <ForwardRef
          className="justify-content-md-center margin0"
          noGutters={false}
        >
          <div
            className="justify-content-md-center margin0 row"
          />
        </ForwardRef>
      </div>
    </div>
  </Home>
</Provider>
`;
